### PR TITLE
(#895) Usage of ScalarWithFallback simplified

### PR DESCRIPTION
**Pull request by scratchboom**
Usage of ScalarWithFallback simplified

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1042](https://github.com/yegor256/cactoos/issues/1042) is now in scope, role is `REV`

File: src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java [hunk 1]
```diff
@@ -58,6 +58,22 @@ public void usesMainFunc() throws Exception {
         );
     }
 
+    @Test
+    public void usesMainFuncInNonIterableCtor() {
+        final String message = "Main function's result";
+        MatcherAssert.assertThat(
+            "Scalar's main function value mismatch",
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: @scratchboom WDYT change this message to something like "Can't match value of Scalar's main function"?

**Comment by fabriciofx** [comment]
@scratchboom Please, fix the PR title.

**Comment by fabriciofx** [comment]
@scratchboom ping

File: src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java [hunk 2]
```diff
@@ -58,6 +58,22 @@ public void usesMainFunc() throws Exception {
         );
     }
 
+    @Test
+    public void usesMainFuncInNonIterableCtor() {
+        final String message = "Main function's result";
+        MatcherAssert.assertThat(
```
[the remark this answers is not in the document]
Comment (scratchboom) [reply src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: Done

File: src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java [hunk 3]
```diff
@@ -96,39 +110,39 @@ public void usesFallbackOfInterruptedException() throws Exception {
...
     public void usesTheClosestFallback() throws Exception {
-        final String expected = "Fallback from IllegalFormatException";
-        MatcherAssert.assertThat(
+        final String expected = "Fallback from IllegalStateException";
+        new Assertion<>(
             "Can't find the closest fallback",
             new ScalarWithFallback<>(
                 () -> {
-                    throw new IllegalFormatWidthException(1);
+                    throw new IllegalStateException("wat is raw");
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: @scratchboom What's "wat"?

Comment (scratchboom) [reply src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: Sorry it's just left from my experiments. This string is not necessary). Removed and pulled.

Comment (fabriciofx) [reply src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: @scratchboom Right.

**Comment by fabriciofx** [comment]
@scratchboom This PR seems fine. Congratulations!

**Comment by fabriciofx** [comment]
@rultor merge

**Comment by scratchboom** [comment]
@llorllale
> you've made some changes that switched exception types around.

That's because I've encountered with "ClassDataAbstractionCouplingCheck" inspection. So I decided to refactor code to minimize the count of used classes by using only some types of Exceptions. (It's doesn't matter which exception classes to use if we test executing the fallback path of ScalarWithFallback).
>Inspection message was something like that - Class Data Abstraction Coupling is 8 (max allowed is 7) classes [Assertion, FallbackFrom, IOException, IllegalFormatWidthException, InterruptedException, IsEqual, IterableOf, ScalarWithFallback]. (ClassDataAbstractionCouplingCheck)

> substituted use of `ScalarHasValue` with `IsEqual`. Why?
As @fabriciofx suggested, I've replaced `MatcherAssert.assertThat` with `new Assertion().affirm()`.
The reason why `IsEqual` used is because `Assertion` receives `Matcher<T>`, but `MatcherAssert.assertThat` receives Matcher<Scalar<T>> in case we make assertion about Scalar.
So I've replaced ScalarHasValue (which is a Matcher<Scalar<T>>) to IsEqual (which is a Matcher<T>).

And actually I've forgot to call `.affirm()` in most places and test obviously were all green. (I've already fixed and pushed this).
I think we should have an inspection, that Assertion instantiation cannot be used without calling `.affirm()`

File: src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java [hunk 4]
```diff
@@ -43,25 +41,41 @@
...
                 new IterableOf<>(
                     new FallbackFrom<>(
-                        new IterableOf<>(IOException.class),
+                        new IterableOf<>(Exception.class),
                         Throwable::getMessage
                     )
                 )
             ),
-            new ScalarHasValue<>(message)
+            new IsEqual<>(message)
```
[the remark this answers is not in the document]
Comment (llorllale) [reply src/test/java/org/cactoos/scalar/ScalarWithFallbackTest.java]: @scratchboom `Assertion.affirm()` is not being called in this test. Also, keep using `ScalarHasValue` and instead change line `ScalarWithFallback` to `() -> ScalarWithFallback(`...

**Comment by llorllale** [comment]
@scratchboom 

> ClassDataAbstractionCouplingCheck

I'm OK with suppressing this in tests.

> The reason why `IsEqual` used is because `Assertion` receives `Matcher<T>`, but `MatcherAssert.assertThat` receives `Matcher<Scalar<T>>` in case we make assertion about Scalar.
So I've replaced ScalarHasValue (which is a `Matcher<Scalar<T>>`) to IsEqual (which is a `Matcher<T>`).

You could've kept using `ScalarHasValue` without any issue. I think the problem is that you are not yet fully aware of how `Assertion` is to be used.

Anyway, left you a comment above.

**Comment by llorllale** [comment]
@scratchboom ping

**Comment by llorllale** [comment]
@scratchboom closing due to inactivity

**Comment by 0crat** [comment]
Code review was too long (36 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#1042](https://github.com/yegor256/cactoos/issues/1042) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Pull request closed without merging**